src/utils.jl

```diff
@@ -1,5 +1,5 @@
 
-const INDEXPARAMS = [:categorical_feature]
+const INDEXPARAMS = [:categorical_feature, :interaction_constraints]
```

Comment:
Extended this to `interaction_constraints` param which requires similar zero-based indexing handling to `categorical_feature`.
There are some dataset params such as `label_column`, `weight_column`, or `group_column` can take a single index indicating the column, or the column name  and these are added in https://github.com/IQVIA-ML/LightGBM.jl/pull/164/files.